src/test/spec/v2_runner/mod.rs

```diff
@@ -134,7 +134,7 @@ pub async fn run_v2_test(test_file: TestFile) {
                 .await;
 
         // TODO RUST-900: Remove this extraneous call.
-        if test.operations.iter().any(|op| op.name == "distinct") {
+        if internal_client.is_replica_set() && internal_client.server_version_lte(4, 2) && test.operations.iter().any(|op| op.name == "distinct") {
```

Comment:
Do we want this to be `is_sharded()` rather than `is_replica_set()`?

Reply:
Yeah 😢  I just noticed that myself, fixing it now.